packages/ui/components/overlays/src/OverlayController.js

```diff
@@ -828,6 +831,12 @@ export class OverlayController extends EventTarget {
       this._hideResolve = resolve;
     });
 
+    // save the current activeElement so we know if the user set focus to another element than the invoker of the dialog
+    // while the dialog was open.
+    // We need this in the _restoreFocus method to determine if we should focus this.elementToFocusAfterHide when the
+    // dialog is closed or keep focus on the element that the user deliberately gave focus
+    this.__activeElement = document.activeElement;
```

Comment:
```suggestion
    this.__activeElement = /** @type {ShadowRoot} */ (this.contentNode.getRootNode()).activeElement;
```
Because document.activeElement only gets the "outer shadow root".

For instance in example below:
```js
<el-in-body>
  <template shadowrootmode="open">
    <lion-dialog>
.     <button id="activeEl"></button>
    </lion-dialog>
  </template>
</el-in-body>
```

document.activeElement will be el-body, but you want to register button#activeEl (else the  "inside dialog check" won't work)

Reply:
Can we maybe call it `__activeElementRightBeforeHide`?

Reply:
Can we maybe call it `__activeElementRightBeforeHide`?